src/test/rpc/TransactionEntry_test.cpp

```diff
@@ -0,0 +1,149 @@
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <test/jtx.h>
+#include <test/jtx/Env.h>
+#include <ripple/protocol/JsonFields.h>
+
+namespace ripple {
+
+class TransactionEntry_test : public beast::unit_test::suite
+{
+    void
+    testBadInput()
+    {
+        testcase("Invalid request params");
+        using namespace test::jtx;
+        Env env {*this};
+
+        {
+            //no params
+            auto const result = env.client()
+                .invoke("transaction_entry", {})[jss::result];
+            BEAST_EXPECT(result[jss::error] == "fieldNotFoundTransaction");
+            BEAST_EXPECT(result[jss::status] == "error");
+        }
+
+        {
+            Json::Value params {Json::objectValue};
+            params[jss::ledger] = 20;
+            auto const result = env.client()
+                .invoke("transaction_entry", params)[jss::result];
+            BEAST_EXPECT(result[jss::error] == "lgrNotFound");
+            BEAST_EXPECT(result[jss::status] == "error");
+        }
+
+        {
+            Json::Value params {Json::objectValue};
+            params[jss::ledger] = "current";
+            params[jss::tx_hash] = "DEADBEEF";
+            auto const result = env.client()
+                .invoke("transaction_entry", params)[jss::result];
+            BEAST_EXPECT(result[jss::error] == "notYetImplemented");
+            BEAST_EXPECT(result[jss::status] == "error");
+        }
+
+        {
+            Json::Value params {Json::objectValue};
+            params[jss::ledger] = "closed";
+            params[jss::tx_hash] = "DEADBEEF";
+            auto const result = env.client()
+                .invoke("transaction_entry", params)[jss::result];
+            BEAST_EXPECT(! result[jss::ledger_hash].asString().empty());
+            BEAST_EXPECT(result[jss::error] == "transactionNotFound");
+            BEAST_EXPECT(result[jss::status] == "error");
+        }
+    }
+
+    void testRequest()
+    {
+        testcase("Basic request");
+        using namespace test::jtx;
+        Env env {*this};
+
+        auto check_tx = [this, &env]
+            (int index, std::string txhash, std::string type = "")
```

Comment:
It'd be cool if this could do `transaction_entry` twice: once with `ledger_index` and again with `ledger_hash`.